tools/run_tests/run_tests.py

```diff
@@ -893,6 +899,37 @@ def runs_per_test_type(arg_str):
         msg = '\'{}\' is not a positive integer or \'inf\''.format(arg_str)
         raise argparse.ArgumentTypeError(msg)
 
+
+def _python_config_generator(name, major, minor, bits, config_args):
+  return PythonConfig(
+    name, config_args['shell'] + config_args['builder'] + config_args['builder_prefix_arguments'] +
+    [_python_pattern_function(major=major, minor=minor, bits=bits)] + [name] + config_args['venv_relative_python'] +
+    config_args['toolchain'], config_args['shell'] + config_args['runner'] +
+    [os.path.join(name, config_args['venv_relative_python'][0])])
+
+
+def _pypy_config_generator(name, major, config_args):
+  return PythonConfig(
+    name, config_args['shell'] + config_args['builder'] + config_args['builder_prefix_arguments'] +
+    [_pypy_pattern_function(major=major)] + [name] + config_args['venv_relative_python'] + config_args['toolchain'],
+    config_args['shell'] + config_args['runner'] + [os.path.join(name, config_args['venv_relative_python'][0])])
+
+
+def _python_pattern_function(major, minor, bits):
+  # Bit-ness is handled by the test machine's environment
+  if os.name == "nt":
+    if bits == "64":
+      return '/c/Python{major}{minor}/python.exe'.format(major=major, minor=minor, bits=bits)
+    else:
+      return '/c/Python{major}{minor}_{bits}bits/python.exe'.format(major=major, minor=minor, bits=bits)
+  else:
+    return 'python{major}.{minor}'.format(major=major, minor=minor)
+
+
+def _pypy_pattern_function(major):
+  return 'pypy{major}'.format(major=major) if major == 3 else 'pypy'
```

Comment:
Depending on how paranoid you're feeling, this might be more safely expressed as
```
if major == 2:
  return 'pypy'
elif major == 3:
  return 'pypy3'
else:
  raise ValueError(<helpful message>)
```
.